File: go/src/practice/longest_string/longest_string.go
package main

import "fmt"

func solution(pString string) (string, int) {
	currLen, maxLen := 0, 0
	var prevLetter rune
	var maxLetter rune
	for _, currLetter := range pString {
		if prevLetter == currLetter {
			currLen++
			if currLen > maxLen {
				maxLen = currLen
				maxLetter = currLetter
			}
		} else {
			currLen = 1
		}
		prevLetter = currLetter
	}
	return string(maxLetter), maxLen
}

func main() {
	str := "AAABBCDAABCCCCDDEEACDD"
	fmt.Println(solution(str))
}

File: python/practice/longest_string.py
def solution(p_string):
    prev_letter, max_letter = None, None
    curr_len = 0
    max_len = 0
    for curr_letter in p_string:
        print(prev_letter, curr_letter)
        if prev_letter == curr_letter:
            curr_len += 1
            if curr_len > max_len:
                max_len = curr_len
                max_letter = curr_letter
        else:
            curr_len = 1
        prev_letter = curr_letter
    return max_len, max_letter

if __name__ == '__main__':
    string = 'AAABBCDAABCCCCDDEEACDD'
    print(solution(string))
